src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,240 @@
+        }
+    }
+
+    /**
+     * ExplosionEvent.Detonate handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Detonate event)
+    {
+        Log.getLogger().debug("ExplosionEvent.Detonate");
+
+        // if block is in colony -> remove from list
+        for (BlockPos pos: event.getAffectedBlocks())
+        {
+            if (colony.isCoordInColony(event.getWorld(), pos))
+            {
+                Log.getLogger().info("Found affected block in colony, removing from affected list");
+                event.getAffectedBlocks().remove(pos);
+            }
+        }
+
+        // if entity is in colony -> remove from list
+        for (Entity entity: event.getAffectedEntities())
+        {
+            if (colony.isCoordInColony(entity.getEntityWorld(), entity.getPosition())) {
+                Log.getLogger().info("Found affected entity in colony, removing from affected list");
+                event.getAffectedEntities().remove(entity);
+            }
+        }
+    }
+
+    /**
+     * PlayerInteractEvent handler.
+     *
+     * Check, if a player right clicked a block.
+     * Deny if:
+     * - If the block is in colony
+     * - block is AbstractBlockHut
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        Log.getLogger().debug("Check if coordinate is in colony `" + this.colony.getName() + "`");
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            Log.getLogger().info("Coordinate is in colony");
+            //todo: check to what hand rightclick maps to
+            //if (event.action == PlayerInteractEvent.Action.RIGHT_CLICK_BLOCK)
+
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+
+            // Huts
+            if (block instanceof AbstractBlockHut &&
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                //todo: hopefully that works
+                event.setResult(Event.Result.DENY);
+            }
+
+        }
+    }
+
+    /**
+     * TODO Check this behavior
+     *
+     * @param event
+     */
+    @SubscribeEvent
+    public void on(final PlayerContainerEvent.Open event)
+    {
+        if (this.colony.isCoordInColony(event.getEntity().getEntityWorld(), event.getEntity().getPosition()))
+        {
+            if (this.colony.getPermissions().isColonyMember(event.getEntityPlayer()))
+            {
+                Log.getLogger().info("Colony member `" + event.getEntityPlayer().getName() + "` opens container `" + event.getEntity().getName() + "`");
+            }
+            else
+            {
+                Log.getLogger().info("Player `" + event.getEntity().getName() + "` is not a member of `" + this.colony.getName() + "`");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Player `" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)